src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+      case 'Circle': {
+        return new OlStyleStyle({
+          fill: new OlStyleFill({
+            color: !useSelectStyle ? fillColor : selectFillColor
+          }),
+          stroke: new OlStyleStroke({
+            color: !useSelectStyle ? strokeColor : selectStrokeColor,
+            width: 2
+          })
+        });
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * Creates a correctly configured OL draw interaction depending on given
+   * drawType and adds this to the map.
+   *
+   * @param {Boolean} pressed Whether the redlining button is pressed or not.
+   * Will be used to handle active state of the draw interaction.
+   *
+   * @method
+   */
+  createDrawInteraction = pressed => {
+    const {
+      drawType,
+      map
+    } = this.props;
+
+    let geometryFunction;
+    let type = drawType;
+
+    if (drawType === 'Rectangle') {
+      geometryFunction = OlInteractionDraw.createBox();
+      type = 'Circle';
+    } else if (drawType === 'Text') {
+      type = 'Point';
+      this._redliningFeatures.on('add', this.handleTextAdding);
+    }
+
+    const drawInteraction = new OlInteractionDraw({
+      source: this.state.redliningLayer.getSource(),
+      type: type,
+      geometryFunction: geometryFunction,
+      style: this.getRedliningStyleFunction,
+      freehandCondition: OlEventsCondition.never
+    });
+
+    map.addInteraction(drawInteraction);
+
+    this.setState({
+      interaction: [drawInteraction]
+    }, () => {
+      drawInteraction.setActive(pressed);
+    });
+  }
+
+  /**
+   * Creates a correctly configured OL select and/or modify and/or translate
+   * interaction(s) depending on given editType and adds this/these to the map.
+   *
+   * @method
+   */
+  createSelectOrModifyInteraction = () => {
+    const {
+      editType,
+      map
+    } = this.props;
+
+    this._selectInteraction = new OlInteractionSelect({
+      condition: OlEventsCondition.singleClick,
+      style: this.getRedliningStyleFunction,
+      hitTolerance: 5
+    });
+
+    if (editType === 'Delete') {
+      this._selectInteraction.on('select', this.onFeatureRemove);
+    } else if (editType === 'Copy') {
+      this._selectInteraction.on('select', this.onFeatureCopy);
+    }
+
+    let interactions = [this._selectInteraction];
+
+    if (editType === 'Edit') {
+      const edit = new OlInteractionModify({
+        features: this._selectInteraction.getFeatures(),
+        deleteCondition: OlEventsCondition.singleClick,
+        style: this.getRedliningStyleFunction,
+        pixelTolerance: 10
+      });
+
+      edit.on('modifystart', this.onModifyStart);
+
+      const translate = new OlInteractionTranslate({
+        features: this._selectInteraction.getFeatures()
+      });
+      interactions.push(edit, translate);
+    }
+
+    interactions.forEach(i => map.addInteraction(i));
+
+    map.on('pointermove', this.onPointerMove);
+
+    this.setState({
+      interaction: interactions
```

Comment:
Is there a need for checking the pressed state of the button in callback (like in the `createDrawInteraction()`)?

Reply:
No, no check needed here